src/Storages/MergeTree/ReplicatedMergeTreeAttachThread.cpp

```diff
@@ -77,6 +82,23 @@ void ReplicatedMergeTreeAttachThread::run()
     }
 }
 
+void ReplicatedMergeTreeAttachThread::checkHasReplicaMetadataInZooKeeper(const zkutil::ZooKeeperPtr & zookeeper, const String & replica_path)
+{
+    /// Since 20.4 and until 22.9 "/metadata" and "/metadata_version" nodes were created on replica startup.
+    /// Since 21.12 we could use "/metadata" to check if replica is dropped (see StorageReplicatedMergeTree::dropReplica),
```

Comment:
```suggestion
    /// Until 21.12 we could use "/metadata" to check if replica is dropped (see StorageReplicatedMergeTree::dropReplica),
```
?

Reply:
No, I mean that code that removes `/metadata` first before removing other nodes was added in 21.12